### ISSUE TITLE
Suggestion for the inclusion of DOE reference and prototype buildings

### ISSUE THREAD

**Issue by borlum** [bsl546/energym#4]
Awesome work! 👍 

For additional buildings (EnergyPlus models) that could be interesting to have available, see:

https://www.energycodes.gov/prototype-building-models
https://www.energy.gov/eere/buildings/commercial-reference-buildings

**Comment by bsl546** [comment]
Hi Joakim, 

Thanks for your comments! 
We are aware of this library of prototype models. For the moment, we are concentrating on Energy models that are entirely designed by members of our Team. Most of them are made to match real buildings and real buildings data. We do not exclude to include the standard prototypes from Energyplus  if their control behavior make sense (I think that now most of them are in closed loop and have no external interfacing for control) . If we see that many users manifest the strong need of having them, then we could include some for sure. Feel free also to participate by trying to include some of them if you want, this project is made so to get as many contributions as possible from the building community!

**Comment by MaPaKo** [comment]
I have integrated one of the "Small Office" models into my local copy of Energym, emulating the "style" of the existing models with temperature setpoints and on/off switches for heat pumps, chillers and gas boilers. So it can be done, but it requires good knowledge of Energym and EnergyPlus.

**Comment by bsl546** [comment]
Yes, sure.  Thanks MaPaKo! Actually if you think  that this particular model is interesting for the community, feel free to contact us, we can create a branch dedicated to its integration into energym and work together on it and on the documentation.

**Comment by MaPaKo** [comment]
I intend to publish the model. Putting it on a new branch here seems like a very good way to do that. 

I chose the small office (version ASHRAE 90.1, 2019, Seattle) because it has a heat pump, unlike the medium office which has a gas boiler. It seems well-insulated, but has low thermal mass by European standards. However, I noticed some odd behavior and asked about it here: 
https://unmethours.com/question/61705/ep-small-office-model-temperature-oscillations/

**Comment by borlum** [comment]
I would be very interested to see what you have done, @MaPaKo - especially in regards to helping with the integration of more of the prototype buildings.

**Comment by MaPaKo** [comment]
I have started to work on a short tutorial, but it will probably be a while until it's finished. First, you have to select the input and output variables you want. This has to be done in three places. You can look at the existing models as a reference. 
- Output Reporting   Output:variable
- External Interface   ExternalInterface:FunctionalMockupUnitExport:From:Variable
- External Interface   ExternalInterface:FunctionalMockupUnitExport:To:Schedule

**Comment by MaPaKo** [comment]
A quick update: I have not made any progress regarding the odd behavior mentioned above. Also, I recently attended a presentation by the developers of EnergyPlus, where it was acknowledged that its idealized room temperature control makes it ill-suited for some types of control development. 

This is why they developed SpawnOfEnergyPlus, which loads EnergyPlus models into Modelica. At some point, I will probably try to integrate such a model into Energym.

**Comment by borlum** [comment]
@MaPaKo, I have successfully compiled FMUs of some of the DOE prototype buildings. However, I am total EnergyPlus noob - how do I know what signals I can set as inputs and outputs? Is there a clever way to browse the EnergyPlus model for this?

**Comment by MaPaKo** [comment]
You can find them in the "Inputs" and "Outputs" sections of the documentation. It takes a while to get into it. Here is an example: 
https://bigladdersoftware.com/epx/docs/9-5/input-output-reference/group-heating-and-cooling-coils.html#outputs-015

PS: I got stuck with other stuff, so it will probably be a while until I can make any of my models available.

**Comment by borlum** [comment]
Thanks for the hints - much appreciated :-)
